### DISCUSSION TITLE
Is it possible to show component in different viewport sizes in MDX Documentation

### DISCUSSION THREAD

**Discussion by matths** [storybookjs/storybook#28449] [Help]
### Summary

Hi!

I really like the .mdx documentation feature of Storybook to show colleagues multiple Components within a single point inside Storybook. Now I would like to also show them the same Story (Component) but in two different viewport sizes, like Mobile VERSUS Desktop.
Is it possible to hand over these kind of parameters in the MDX or what would you do to do this?

Let's start with this example demo.mdx File:

```
import { Meta, Canvas } from '@storybook/blocks';

import * as DemoStories from './Demo.stories.svelte';

# Demo

I can put my documentation here as Markdown

<Meta title="Tree/Path/Story" />
<Canvas
  of={DemoStories.myDemo}
  withToolbar="true"
  source={{
    code: `// My custom source`,
  }}
  sourceState="shown"
/>
```

Can I somehow add two Canvas, one in Desktop size, one in Mobile size?

Cheers,
Matthias

### Additional information

```
    "@storybook/addon-essentials": "^8.1.11",
    "@storybook/addon-interactions": "^8.1.11",
    "@storybook/addon-links": "^8.1.11",
    "@storybook/addon-options": "5.3.21",
    "@storybook/addon-svelte-csf": "4.1.3",
    "@storybook/addon-viewport": "^8.1.11",
    "@storybook/blocks": "^8.1.11",
    "@storybook/svelte": "^8.1.11",
    "@storybook/svelte-vite": "^8.1.11",
    "@storybook/test": "^8.1.11",
```

**Comment by matths** [comment]
I still couldn't figure this out. Do you know, where I could put this question instead?

**Reply by andreszapatavff** [reply]
@matths any luck?

**Reply by matths** [reply]
well, kind of. My workaround uses iframes in the docs referring to Stories, but I had to add fixed width to get my media queries to be applied.

```markdown
### Mobile Viewport:

<iframe
  src="?path=/story/my-components--demo&globals=viewport%3AcustomMobile"
  style={{
    width: '375px',
    height: '480px',
    border: '1px solid #ccc',
  }}
/>

### Desktop Viewport:

<iframe
  src="?path=/story/my-components--demo&globals=viewport%3AcustomDesktop"
  style={{
    width: '860px',
    height: '480px',
    border: '1px solid #ccc',
  }}
/>

```

Let me know if that is of any use or if you find other/better solutions.
